process/block/shardblock.go

```diff
@@ -141,7 +141,8 @@ func (sp *shardProcessor) ProcessBlock(
 				"for shard", headerHandler.GetShardID(),
 			)
 
-			go sp.requestHandler.RequestShardHeader(headerHandler.GetShardID(), headerHandler.GetPrevHash())
+			// TODO : check epoch here
+			go sp.requestHandler.RequestShardHeader(headerHandler.GetShardID(), headerHandler.GetPrevHash(), headerHandler.GetEpoch())
```

Comment:
as explained above, no way to know epoch.

Reply:
refactored